R/test-threads.R

```diff
@@ -97,7 +97,15 @@ test_threads <- function(
     writeLines(mod_lines, mod_path)
   })
 
-  submit_models(.mods, .wait = FALSE, ...)
+
+  tryCatch({
+    submit_models(.mods, .wait = FALSE, ...)
+  },
+  error = function(cond){
+    delete_models(.mods = .mods, .force = TRUE) %>% suppressMessages()
+    message(cond)
+    return(invisible(TRUE))
```

Comment:
Why return `TRUE`?  I expected this to clean up the models and then re-signal the error.  If there's some reason it shouldn't do that, then I'd say `NULL` would be the more typical return value.

Reply:
It does re-signal the error via `message(cond)`, but the important change was deleting the newly created model files _if_ submission fails. Im fine with changing it to `NULL` though.

Reply:
> It does re-signal the error via `message(cond)`

That doesn't re-signal it; it gives a message about the error.  Re-signaling it would mean that execution was halted unless outer code caught it.

This...

```R
f1 <- function() {
  stop("boom")
}

f2 <- function() {
  tryCatch(f1(),
           error = function(e) {
             message(e, " is the message")
           })
}

f2()
message("on I go")
```

versus this:

```diff
diff --git a/scratch.R b/scratch-rethrow.R
index 2952a0c..e0a1533 100644
--- a/scratch.R
+++ b/scratch-rethrow.R
@@ -6,7 +6,7 @@
 f2 <- function() {
   tryCatch(f1(),
            error = function(e) {
-             message(e, " is the message")
+             stop(e)
            })
 }
```

Perhaps demoting the error to a message is the right thing to do here, but it's different (and, as I said, not what I expected).  Is there a reason you prefer to give the message rather than propagate the error?

Reply:
Ahh I take your point. And no to be honest most of the trycatch examples I had looked at in the past used message, so thats probably where I got it from. Can make the change